pkg/binding/controller.go

```diff
@@ -267,6 +280,23 @@ func (c *Controller) run(ctx context.Context, workers int) error {
 	return nil
 }
 
+func (c *Controller) createManagedClustersInformer(ctx context.Context) {
+	informerFactory := ocm.GetOCMInformerFactory(c.ocmClientset)
+	informer := informerFactory.Cluster().V1().ManagedClusters().Informer()
+	_, err := informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		// TODO: we can be smarter on how we handle at least the delete rather than
+		// recreate all the bindings.
+		AddFunc:    func(obj interface{}) { c.evaluateBindingPolicies(ctx) },
+		UpdateFunc: func(oldObj, newObj interface{}) { c.evaluateBindingPolicies(ctx) },
+		DeleteFunc: func(obj interface{}) { c.evaluateBindingPolicies(ctx) },
```

Comment:
in delete function you can check the labels listed on the MC.
then when you get the placement objects, you can check if the cluster selector uses any of the labels above. if not, it's not relevant to handle that placement.
so you need to handle only placements that include a label that the deleted MC had.

Reply:
Why only on the delete and not also on an add?  For update, we can do the same for any placement object that has a cluster selector that use either the old obj or the new obj labels.

Reply:
on add as well (I wrote it in the next comment). for update I would be careful but basically you're correct. you should only care about placements that include any label that exist in the `Union(oldObj.labels, newObj.labels)`.

Reply:
Done.

Reply:
@nirrozenbaum those are two different controllers, can't we have race conditions here ? i.e., the label is not there when you check and but in practice it was already added

Reply:
Can you be more precise @ezrasilvera and give an exact scenario?  Perhaps I can add it to the test.  K8s is eventual consistency by nature.

Reply:
@ezrasilvera I'm not sure I understood the race. 
there is **one**  controller that watches clusters and BindingPolicy (and all workload objects but let's put that aside). can you try to describe such a race?